src/Compilers/Core/Portable/Operations/OperationWalker.cs

```diff
@@ -54,4 +54,57 @@ internal override void VisitNoneOperation(IOperation operation)
             VisitArray(operation.Children);
         }
     }
+
+    /// <summary>
+    /// Represents a <see cref="OperationVisitor{TArgument, TResult}"/> that descends an entire <see cref="IOperation"/> tree
+    /// visiting each IOperation and its child IOperation nodes in depth-first order. Returns null.
+    /// </summary>
+    public abstract class OperationWalker<TArgument> : OperationVisitor<TArgument, object?>
+    {
+        private int _recursionDepth;
+
+        internal void VisitArray<T>(IEnumerable<T> operations, TArgument argument) where T : IOperation
+        {
+            foreach (var operation in operations)
+            {
+                VisitOperationArrayElement(operation, argument);
+            }
+        }
+
+        internal void VisitOperationArrayElement<T>(T operation, TArgument argument) where T : IOperation
```

Comment:
>VisitOperationArrayElement [](start = 22, length = 26)

Is presence of this method necessary? It doesn't look like it is doing anything interesting or complicated.